x/ibc/20-transfer/types/packet.go

```diff
@@ -62,43 +63,43 @@ func (pd PacketDataTransfer) ValidateBasic() error {
 	return nil
 }
 
-// Implements channelexported.PacketDataI
+// GetBytes implements channelexported.PacketDataI
 func (pd PacketDataTransfer) GetBytes() []byte {
 	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(pd))
 }
 
-// Implements channelexported.PacketDataI
+// GetTimeoutHeight implements channelexported.PacketDataI
 func (pd PacketDataTransfer) GetTimeoutHeight() uint64 {
 	return pd.Timeout
 }
 
-// Implements channelexported.PacketDataI
+// Type implements channelexported.PacketDataI
 func (pd PacketDataTransfer) Type() string {
 	return "ics20/transfer"
 }
 
 var _ channelexported.PacketDataI = AckDataTransfer{}
 
-type AckDataTransfer struct {
-}
+// AckDataTransfer is a no-op packet
+// See spec for onAcknowledgePacket: https://github.com/cosmos/ics/tree/master/spec/ics-020-fungible-token-transfer#packet-relay
+type AckDataTransfer struct{}
```

Comment:
shouldn't this packet type have like a field for the hash of the packet it's acknowledging?

Reply:
It doesn't matter because the packet is uniquely identified by its sequence number (plus the port & channel).

Reply:
However, we may want to deal with the case of a *failed* transfer (e.g. receiving account does not exist), and if we do that should be handled in the acknowledgement.

Reply:
@cwgoes is there spec for that?

Reply:
I am still not sure if this (transfer failing) is possible or not in the SDK, which will determine whether or not we need to handle it. Glad to write a spec if we need one!

Reply:
Can `sdk.AddCoins` fail?

Reply:
[`handleRecvPacket`](ttps://github.com/cosmos/cosmos-sdk/blob/ibc-alpha/x/ibc/20-transfer/handler.go#L59) calls -> [`SendCoinsFromModuleToAccount`]( https://github.com/cosmos/cosmos-sdk/blob/ibc-alpha/x/ibc/20-transfer/keeper/relay.go#L79) calls -> [`supply.SendCoins`](https://github.com/cosmos/cosmos-sdk/blob/ibc-alpha/x/supply/internal/keeper/bank.go#L22) calls -> [`SubtractCoins`](https://github.com/cosmos/cosmos-sdk/blob/ibc-alpha/x/bank/internal/keeper/keeper.go#L235) which returns an error. I think there are also a couple of other paths here to a failed `AddCoins`